homeassistant/components/irm_kmi/const.py

```diff
@@ -0,0 +1,109 @@
+"""Constants for the IRM KMI integration."""
+
+from typing import Final
+
+from homeassistant.components.weather import (
+    ATTR_CONDITION_CLEAR_NIGHT,
+    ATTR_CONDITION_CLOUDY,
+    ATTR_CONDITION_FOG,
+    ATTR_CONDITION_LIGHTNING_RAINY,
+    ATTR_CONDITION_PARTLYCLOUDY,
+    ATTR_CONDITION_POURING,
+    ATTR_CONDITION_RAINY,
+    ATTR_CONDITION_SNOWY,
+    ATTR_CONDITION_SNOWY_RAINY,
+    ATTR_CONDITION_SUNNY,
+)
+from homeassistant.const import Platform, __version__
+
+DOMAIN: Final = "irm_kmi"
+PLATFORMS: Final = [Platform.WEATHER]
+CONFIG_FLOW_VERSION = 1
+
+OUT_OF_BENELUX: Final = [
+    "außerhalb der Benelux (Brussels)",
+    "Hors de Belgique (Bxl)",
+    "Outside the Benelux (Brussels)",
+    "Buiten de Benelux (Brussel)",
+]
+LANGS: Final = ["en", "fr", "nl", "de"]
+
+CONF_LANGUAGE_OVERRIDE: Final = "language_override"
+
+CONF_LANGUAGE_OVERRIDE_OPTIONS: Final = ["none", "fr", "nl", "de", "en"]
+
+REPAIR_SOLUTION: Final = "repair_solution"
+REPAIR_OPT_MOVE: Final = "repair_option_move"
+REPAIR_OPT_DELETE: Final = "repair_option_delete"
+REPAIR_OPTIONS: Final = [REPAIR_OPT_MOVE, REPAIR_OPT_DELETE]
+
+# Dict to map ('ww', 'dayNight') tuple from IRM KMI to HA conditions.
+IRM_KMI_TO_HA_CONDITION_MAP: Final = {
+    (0, "d"): ATTR_CONDITION_SUNNY,
+    (0, "n"): ATTR_CONDITION_CLEAR_NIGHT,
+    (1, "d"): ATTR_CONDITION_SUNNY,
+    (1, "n"): ATTR_CONDITION_CLEAR_NIGHT,
+    (2, "d"): ATTR_CONDITION_LIGHTNING_RAINY,
+    (2, "n"): ATTR_CONDITION_LIGHTNING_RAINY,
+    (3, "d"): ATTR_CONDITION_PARTLYCLOUDY,
+    (3, "n"): ATTR_CONDITION_PARTLYCLOUDY,
+    (4, "d"): ATTR_CONDITION_POURING,
+    (4, "n"): ATTR_CONDITION_POURING,
+    (5, "d"): ATTR_CONDITION_LIGHTNING_RAINY,
+    (5, "n"): ATTR_CONDITION_LIGHTNING_RAINY,
+    (6, "d"): ATTR_CONDITION_POURING,
+    (6, "n"): ATTR_CONDITION_POURING,
+    (7, "d"): ATTR_CONDITION_LIGHTNING_RAINY,
+    (7, "n"): ATTR_CONDITION_LIGHTNING_RAINY,
+    (8, "d"): ATTR_CONDITION_SNOWY_RAINY,
```

Comment:
how does this work?

Reply:
It's a mapping between Home Assistant and the API. The number refers to the type and is being returned by the API and d and n refer to day and night. The full list is documented on the hacs repository: https://github.com/jdejaegh/irm-kmi-ha